Document Packit-as-a-Service (#15)
packit-service/packit.dev: content/packit-as-a-service.md

---
title: "Packit-as-a-Service"
date: 2019-06-22
draft: false
---

> Packit-as-a-Service is a GitHub Application (https://developer.github.com/apps/about-apps/).

Reviewer: ```suggestion
Packit-as-a-Service is a GitHub App (https://developer.github.com/apps/about-apps/).
```

### Easy to use

> Using Packit Service is very straightforward: add one config file to your repository,

Reviewer: a link do docs please

---

> title: "Packit-as-a-Service"

Reviewer 2: can we name it Documentation?

Author: It makes sense to me if `packit.dev` is web only about `Packit-as-a-Service`. But If we talk about `Packit` and `Packit-as-a-Service`, then I wanted to differentiate it a bit. That was my proposal.

Reviewer 2: Yes, that's a good point. In that case, packit-as-a-service makes sense.

On the flip side, I'd say that our end-goal is packit-as-a-service where packit tool is really just a detail of the complete solution.

Oki, let's leave it as it is.

date: 2019-06-22
draft: false
---

> ## Introductory description

Reviewer 2: this line is not needed really

With packit, it's easy to integrate your upstream project into Fedora Project.
Your pull requests are built in Fedora version of your choice,
so you're always sure that your software works. When you're ready for a release,
packit takes care of all the hassle and is able to ship your new upstream release

> into Fedora Rawhide.

Author: - Pull Requests are built in Fedora version, or more correctly, code in pull requests is built in Fedora version.
- `packit takes care` is meant to be CLI. `Packit-as-a-Service` is built on the top of packit CLI tool. The code is performed by `packit`

But I would remove `Introductory description` at all. After a time, I don't see any advantage to have Marketing pitch there.

### Easy to use

Using Packit Service is very straightforward: add one config file to your repository,
followed by an RPM spec file and you're good to go. We also support packit as a CLI tool,

> so you can always try things locally on your own.

Author: After reading several times. `packit as a tool` is also useful. `CLI tool` looks weird.

> so you can always try things locally on your own.

Reviewer: Ok, what I meant was: This doc is about the GitHub App, so why mention the tool/CLI at all?

We can leave it here, but with a little rewording and link to the tool docs.
"Besides the service, Packit is also a tool/CLI [a link to some doc]"

### Makes you confident

Packit Service validates your pull requests by building your software in Fedora OS.
Once the builds are done, packit lets you know how to install your change
inside your environment. Get confidence by trying out your changes yourself

> before shipping them to your users.

Reviewer 2: I'm not sure we need a marketing pitch in the intro/docs, how about putting it to "About"?

Author: Yeah, it could be. I will move it.

Author: Does it make sense to create in "About" section two subsections like:

~~~
$ About.md
---
title:
---
## Packit-as-a-Service
marketing

## Packit
info about packit
~~~

Reviewer: I wouldn't pollute "About" with this.
I think Tomas just wanted to say that we don't need such marketing tone.
What I'd suggest:
- remove `### Makes you confident` to merge this with the previous section
- remove the `Get confidence` and merge the sentences into `Once the builds are done, packit lets you know how to install the RPMs inside your environment so you can try the changes before merging them.`

Author: ok, I would prefer to remove it as well completely. Waiting for other opinions.

Reviewer: > Waiting for other opinions.

there's no one else, we are all alone here :-)

Author: Like @TomasTomecek. He already reacts by emoji.

## Steps how to install Packit-as-a-Service into your projects or organizations

There are two ways to install Packit Service GitHub app into your repository.

### Install from Packit-as-a-Service page
1. From the [Packit-service page](https://github.com/apps/packit-as-a-service) page,
in the left sidebar, click "Install App"
2. Click "Install" next to the organization or user account containing the correct repository

> 3. Install the "Packit-as-a-Service" on all repositories or select repositories

Author: I have changed to `or to certain repositories`

---
title: "Packit-as-a-Service"
date: 2019-06-22
draft: false
---

Packit-as-a-Service is a GitHub App (https://developer.github.com/apps/about-apps/).

### Easy to use

Using Packit Service is very straightforward: add one config file to your repository
([see .packit.yaml format](https://github.com/packit-service/packit/blob/master/docs/configuration.md)),

> followed by an RPM spec file and you're good to go. Besides the service, packit is also as a tool

Reviewer 2: ```suggestion
followed by an RPM spec file and you're good to go. Besides the service, packit is also a tool
```

([see packit README.md](https://github.com/packit-service/packit/blob/master/README.md)),
so you can always try things locally on your own.

Packit Service validates your pull requests by building your software in Fedora OS.
Once the builds are done, packit lets you know how to install the
RPMs inside your environment so you can try the changes before merging them.

<!--more-->

## Steps how to integrate Packit-as-a-Service into your projects or organizations

There are two ways to install Packit-as-a-Service GitHub app into your repository.

### Install from Packit-as-a-Service page
1. From the [Packit-service page](https://github.com/apps/packit-as-a-service) page,
in the left sidebar, click "Install App"
2. Click "Install" next to the organization or user account containing the correct repository
3. Install the "Packit-as-a-Service" on all repositories or to certain repositories

> ### Install from GitHub Marketplace

Reviewer 2: since we won't be on marketplace in following days, I'd comment this section out

1. Go to [GitHub Marketplace](https://github.com/marketplace)
2. In section [Continuous integration](https://github.com/marketplace?category=continuous-integration)
find "Packit-as-a-Service" and select it
3. On the "Packit-as-a-Service" page, under "Pricing and setup", click "Install it for free"
4. Click "Complete order and begin installation"
5. Install the "Packit-as-a-Service" on all repositories or to certain repositories

Once installed, you will see "Packit-as-a-Service" GitHub application in your project settings.
In the left sidebar, click "Integration & services" and our application is shown here.

### Add .packit.yaml configuration file

Packit Service expects ".packit.yaml" configuration file in root of your upstream repository.
For more info
[see .packit.yaml documentation](https://github.com/packit-service/packit/blob/master/docs/configuration.md)

### Add SPEC file

Packit Service needs RPM spec file in your upstream repository.

> This RPM spec file is used for building project in COPR repositories.

Reviewer 2: ```suggestion
This RPM spec file is used to build your project.
```

> If this makes you barf, you can use a “source git” branch of your upstream repository.

Reviewer 2: WHAT?!

Author: I have added there from one of our document. Will remove it and RFE remains.

Packit is also tracking an RFE to not require spec in upstream
[packit-service/packit#312](https://github.com/packit-service/packit/issues/312).

### How to check Packit Service works in your project?

> Nowadays, Packit Service supports only validation of your pull requests.

Reviewer 2: ```suggestion
Nowadays, Packit Service is able only to build your changes and provide them as RPMs.
```
validation implies testing

Once installed, you will see "Packit-as-a-Service" GitHub application in your project settings.
In the left sidebar, click "Integration & services" and our application is shown here.

### Add .packit.yaml configuration file

Packit Service expects ".packit.yaml" configuration file in root of your upstream repository.
For more info
[see .packit.yaml documentation](https://github.com/packit-service/packit/blob/master/docs/configuration.md)

### Add SPEC file

> Packit Service needs RPM spec file in your upstream repository. This RPM spec file is used for building

Reviewer: Could you link https://github.com/packit-service/packit/issues/312 here?
